packer/plugin-getter/github/getter.go

```diff
@@ -156,10 +158,40 @@ func (t *HostSpecificTokenAuthTransport) base() http.RoundTripper {
 	return http.DefaultTransport
 }
 
+type GithubPlugin struct {
+	Hostname  string
+	Namespace string
+	Type      string
+}
+
+func PluginToGithubURI(source *addrs.Plugin) (*GithubPlugin, error) {
```

Comment:
```suggestion
func NewGitHubPluginURI(source *addrs.Plugin) (GithubPlugin, error) {
```

I think you can also just take in three arguments here (hostname, namespace, reponame string) and drop the validation for three parts and the passing of an addrs.Plugin. Or just take in a string and split on to validate three parts.

Reply:
We could take three arguments, but in the end the splitting into three parts will need to be done at callsite instead of in that function, I'm not sure this is better?
Taking the source string works as well, but since the `addrs.Plugin` type offers a `Parts()` function that does this well for us, making this work on any string will essentially involve re-doing that code locally, so I'm not certain of this alternative. (Besides maybe testing? Although it's easy to make a plugin with a source string in that context).